From 99a35fc92be2c1d04b1dae728b40addbe92bf930 Mon Sep 17 00:00:00 2001
From: Phil Rosenberg <p.d.rosenberg@gmail.com>
Date: Tue, 15 Dec 2015 00:27:31 +0000
Subject: [PATCH] Only Redraw wxStaticBox if it has non-zero size

If a wxStaticBox is rendered, but has zero size in one dimention we
generate assert failures when creating the bitmap. Check for these
conditions and just do not render if this is the case.
---
 src/msw/statbox.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/msw/statbox.cpp b/src/msw/statbox.cpp
index b3755c9..b271b96 100644
--- a/src/msw/statbox.cpp
+++ b/src/msw/statbox.cpp
@@ -507,6 +507,12 @@ void wxStaticBox::OnPaint(wxPaintEvent& WXUNUSED(event))
     ::GetClientRect(GetHwnd(), &rc);
     wxPaintDC dc(this);
 
+	// no need to do anything if rc has zero size in either dimension
+	if ( rc.left == rc.right )
+		return;
+	if (rc.top == rc.bottom)
+		return;
+
     // draw the entire box in a memory DC
     wxMemoryDC memdc(&dc);
     wxBitmap bitmap(rc.right, rc.bottom);
-- 
2.5.3

